tests/actions/IOUTest.js

```diff
@@ -0,0 +1,956 @@
+
+                            // There should be one transaction
+                            expect(_.size(allTransactions)).toBe(1);
+                            const transaction = _.find(allTransactions, t => !_.isEmpty(t));
+                            transactionID = transaction.transactionID;
+
+                            expect(transaction.reportID).toBe(iouReportID);
+                            expect(transaction.amount).toBe(amountInCents);
+                            expect(transaction.comment.comment).toBe(comment);
+                            expect(transaction.merchant).toBe(CONST.REPORT.TYPE.IOU);
+                            expect(transaction.pendingAction).toBe(CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+
+                            // The transactionID on the iou action should match the one from the transactions collection
+                            expect(iouAction.originalMessage.IOUTransactionID).toBe(transactionID);
+
+                            resolve();
+                        },
+                    });
+                }))
+                .then(() => {
+                    fetch.fail();
+                    return fetch.resume();
+                })
+                .then(() => new Promise((resolve) => {
+                    const connectionID = Onyx.connect({
+                        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReportID}`,
+                        waitForCollectionCallback: true,
+                        callback: (reportActionsForChatReport) => {
+                            Onyx.disconnect(connectionID);
+                            expect(_.size(reportActionsForChatReport)).toBe(2);
+                            iouAction = _.find(reportActionsForChatReport, reportAction => reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU);
+                            expect(iouAction.pendingAction).toBeFalsy();
+                            const errorMessage = _.values(iouAction.errors)[0];
+                            expect(errorMessage).toBe(Localize.translateLocal('iou.error.genericCreateFailureMessage'));
+                            resolve();
+                        },
+                    });
+                }))
+                .then(() => new Promise((resolve) => {
+                    const connectionID = Onyx.connect({
+                        key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+                        waitForCollectionCallback: true,
+                        callback: (transaction) => {
+                            Onyx.disconnect(connectionID);
+                            expect(transaction.pendingAction).toBeFalsy();
+                            expect(transaction.errors).toBeTruthy();
+                            expect(_.values(transaction.errors)[0]).toBe(Localize.translateLocal('iou.error.genericCreateFailureMessage'));
+
+                            // Cleanup
+                            fetch.succeed();
```

Comment:
Just a preference, move this to another chained promise.

Reply:
done